docs/proposals/PROPOSAL_I.md

```diff
@@ -0,0 +1,235 @@
+
+    - **`variant`** _string_
+
+    - **`os`** _string_
+
+    - **`os.version`** _string_
+
+    - **`os.features`** _array of strings_
+
+    - **`features`** _array of strings_
+
+    - **`compatibilities`** _[descriptor](https://github.com/opencontainers/image-spec/blob/main/descriptor.md)_
+
+      This OPTIONAL property specifies a descriptor of a compatibilities manifest.
+
+## Compatibilities Manifest
+
+The compatibilities manifest is a special image manifest, contains only config and no blob layer.
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "config": {
+    "mediaType": "application/vnd.oci.image.config.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 678
+  },
+  "layers": [],
+  "annotations": {
+    "oci.opencontainers.image.created": "2024-03-27T08:08:08Z"
+  }
+}
+```
+
+### Config Extention
+
+The config object will be extened with compatibilities field.
+
+The required rootfs COULD be empty if the config contains compatibilities field.
+
+- **`compatibilities`** _array of compatibility sets_
+
+  This REQUIRED property describes all image compatibility sets, contains at least one compatibility set.
+
+  - **`compatibility set`** _array of compatibility requirements_
+
+    This REQUIRED property describes compatibility requirements in one compatibility set.
+
+    If one of the compatibility sets passes the compatibility check against the target node, that means the node is compatible with the image.
+
+    There are three category key-value pair inside the compatibility set.
+
+    - **`compatibility label`** _string key value pair_
+
+      This REQUIRED property describes a feature compatibility requirement on the target host, the label accepts OCI and community defined labels.
+
+      OCI will only maintain standardized compatility lable proposed by communities.
+
+      The value of compatibility label SHOULD be described in contiguous or discontinuous range if the value discovered on nodes can be different.
+
+      It's the tool's responsibility to interpret the value of OCI or community defined label, and matching criteria with the value discovered from the node via NFD tool or other tool. NFD means the Kubernetes node feature discovering SIG.
+
+      Inevitably, some compatibility fields MAY be duplicated among compatibility sets.
+
+      The value type of this field is STRING.
+
+    - **`tags`** _array of string_
+
+      This OPTIONAL property allows to group compatibility sets by tags.
+
+      The value type of this field is STRING.
+
+    - **`description`**
+
+      This OPTIONAL property is to describe human-readable information on a compatibility set.
+
+      The value type of this field is STRING.
+
+### Config Example
+
+All labels in this section are only for examples.
+
+#### Example - image is compatible with different hardware combinations
+
+```json
+{
```

Comment:
This looks like just slightly changing the format / structure to look like a config over an artifact. There are still many jumps to get here - I'm not sure why this approach is better? It has more down sides, like requiring the person own the image / registry URI to be able to update or define the compatibilities. And then the issue of changing digests.

Reply:
for jump, it's similar as artifact.

> It has more down sides, like requiring the person own the image / registry URI to be able to update or define the compatibilities. And then the issue of changing digests.

it's quite easy to update compatibilites, just replace the descriptor in the image index manifest entry. and the image itself doesn't need to update , because the image manifest digest didnot change yet. it's very flexible while keep current flow unchanged.